prov/gni/src/gnix_rma.c

```diff
@@ -1478,6 +1485,9 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 			_gnix_vc_queue_tx_req(more_req);
 			slist_init(&ep->more_read);
 		}
+		if (flags & FI_FENCE) {
```

Comment:
Can you add a comment above this explaining why you need it.  FI_FENCE is a bit tricky to follow through here.